lib/UI/calendar/widgets/date_widget.dart

```diff
@@ -35,9 +37,7 @@ class DateWidget extends StatelessWidget {
     return StreamBuilder<QuerySnapshot<Map<String, dynamic>>>(
       stream: getWorkoutOnDate(after, before),
       builder: (context, snapshot) {
-        if (snapshot.hasError) {
-          return _buildDayContainer();
-        } else if (snapshot.hasData) {
+        if (snapshot.hasData) {
```

Comment:
Wouldn't a guard clause be clearer here? seeing as the else function only is there to return _buildDayContainer(). Makes it clearer to read in my opinion.

Reply:
> Wouldn't a guard clause be clearer here? seeing as the else function only is there to return _buildDayContainer(). Makes it clearer to read in my opinion.

Everything looks good, only this comment left. Are you still working on this or don't you agree?

Reply:
I think I misunderstood. Can you explain what you mean by guard clause? I already modified the file like this:

Reply:
with guard clause I mean something like this.

```
if (!snapshot.hasData) return _buildDayContainer();

Workout? workout;
try {
  workout = Workout.fromJsonQuerySnapshot(snapshot.data!);
} catch (_) {
  /// This results in bad state, due to parsing error
  // Log.error(e.toString(), stackTrace: stackTrace);
}
...
```

Makes it a little clearer to read because you don't have to put all your code inside the if statement.

Reply:
Alright, gotcha.